acme/acme.go

```diff
@@ -112,10 +127,36 @@ func (a *ACME) init() error {
 		log.Warn("ACME.StorageFile is deprecated, use ACME.Storage instead")
 		a.Storage = a.StorageFile
 	}
+
+	if len(a.DNSProvider) > 0 {
+		log.Warn("ACME.DNSProvider is deprecated, use ACME.DNSChallenge instead")
+		a.DNSChallenge = &DNSChallenge{DNSProvider: a.DNSProvider, DelayDontCheckDNS: a.DelayDontCheckDNS}
+	}
+
+	if a.OnDemand {
+		log.Warn("ACME.OnDemand is deprecated")
+	}
+
```

Comment:
I would use `SetEffectiveConfiguration()` to initialize the configuration correctly instead of `init()`